crates/common_utils/src/pii.rs

```diff
@@ -24,6 +25,103 @@ pub const REDACTED: &str = "Redacted";
 /// Type alias for serde_json value which has Secret Information
 pub type SecretSerdeValue = Secret<serde_json::Value>;
 
+/// Phone Number
+#[derive(Debug, serde::Deserialize, serde::Serialize)]
+pub struct PhoneNumber(Secret<String>);
+
+impl<T> Strategy<T> for PhoneNumber
+where
+    T: AsRef<str> + std::fmt::Debug,
+{
+    fn fmt(val: &T, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let val_str: &str = val.as_ref();
+
+        if val_str.len() < 10 || val_str.len() > 12 {
+            return WithType::fmt(val, f);
+        }
+
+        write!(
+            f,
+            "{}{}{}",
+            &val_str[..2],
+            "*".repeat(val_str.len() - 5),
+            &val_str[(val_str.len() - 3)..]
+        )
+    }
+}
+
+impl FromStr for PhoneNumber {
+    type Err = error_stack::Report<ValidationError>;
+    fn from_str(phone_number: &str) -> Result<Self, Self::Err> {
+        validate_phone_number(phone_number)?;
+        let secret = Secret::<String>::new(phone_number.to_string());
+        Ok(Self(secret))
+    }
+}
+
+impl TryFrom<String> for PhoneNumber {
+    type Error = error_stack::Report<errors::ParsingError>;
+
+    fn try_from(value: String) -> Result<Self, Self::Error> {
+        Self::from_str(&value).change_context(errors::ParsingError::EmailParsingError)
+    }
+}
+
+// From should not be available because it may create an invalid PhoneNumber object
+
+// impl From<Secret<String>> for PhoneNumber {
+//     fn from(value: Secret<String>) -> Self {
+//         Self(value)
+//     }
+// }
+
+impl ops::Deref for PhoneNumber {
+    type Target = Secret<String>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl ops::DerefMut for PhoneNumber {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.0
+    }
+}
+
+impl<DB> Queryable<diesel::sql_types::Text, DB> for PhoneNumber
+where
```

Comment:
Wondering whether we can reduce this repetition associated with NewTypes...
maybe we can create a generic trait or a macro_rules....

We could take this up in the future maybe
@SanchithHegde should we create an issue for this^

Reply:
Yeah we can create an issue for this.